### PR TITLE
Add Mongo CommandListener rule

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
Adds Byteman rules and tests for https://github.com/opentracing-contrib/java-mongo-driver auto-instrumentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,6 +136,11 @@ Currently supported containers:
 Framework integration repo: [OkHttp](https://github.com/opentracing-contrib/java-okhttp)
 
 
+### Mongo Driver
```
Comment (objectiser) [inline README.md]: This needs to be one level down, e.g. `####`

File: rules/java-mongo-driver/pom.xml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    <dependency>
+      <groupId>io.opentracing</groupId>
+      <artifactId>opentracing-mock</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>io.opentracing.contrib</groupId>
+      <artifactId>opentracing-mongo-driver</artifactId>
+      <version>0.0.6</version>
```
Comment (objectiser) [inline rules/java-mongo-driver/pom.xml]: Can you define the version in the root pom.

File: rules/java-mongo-driver/pom.xml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    <artifactId>opentracing-agent-rules-parent</artifactId>
+    <groupId>io.opentracing.contrib</groupId>
+    <version>0.4.1-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>opentracing-agent-rules-java-mongo-driver</artifactId>
+
+  <properties>
+    <opentracing-agent.lib>${project.build.directory}/lib</opentracing-agent.lib>
+    <mongo.driver.version>3.8.1</mongo.driver.version>
```
Comment (objectiser) [inline rules/java-mongo-driver/pom.xml]: Could you move this property to the root pom.

File: rules/java-mongo-driver/pom.xml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>${version.junit}</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>de.bwaldvogel</groupId>
+      <artifactId>mongo-java-server</artifactId>
+      <version>1.8.0</version>
```
Comment (objectiser) [inline rules/java-mongo-driver/pom.xml]: Can you move this version to a property in the root pom.

File: rules/java-mongo-driver/src/test/java/io/opentracing/contrib/agent/mongo/AsyncMongoClientITest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates
```
Comment (objectiser) [inline rules/java-mongo-driver/src/test/java/io/opentracing/contrib/agent/mongo/AsyncMongoClientITest.java]: Could you change to `Copyright 2018 The OpenTracing Authors`.

File: rules/java-mongo-driver/src/test/java/io/opentracing/contrib/agent/mongo/MongoClientITest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates
```
Comment (objectiser) [inline rules/java-mongo-driver/src/test/java/io/opentracing/contrib/agent/mongo/MongoClientITest.java]: Same as above.

**Review by objectiser — COMMENTED** [review]
Thanks for this - only some minor changes required.

**Comment by rmfitzpatrick** [comment]
@objectiser thanks for the review.  I believe I've addressed everything, but it looks like I should also add the introduced dependencies to the root pom w/ compile scope as the others have done?

**Comment by objectiser** [comment]
@rmfitzpatrick Yes please

**Comment by rmfitzpatrick** [comment]
@objectiser, appreciate the help.  Updated.

**Comment by objectiser** [comment]
@rmfitzpatrick Thanks!

**Comment by objectiser** [comment]
Released in 0.5.0.

**Pull request merged**